rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java

```diff
@@ -1161,6 +1165,32 @@ public String getJobHtml(@HeaderParam("sessionid") String sessionId, @PathParam(
         }
     }
 
+    @Override
+    @GET
+    @Path("jobs/{jobid}/xml")
+    @Produces("application/xml")
+    public InputStream getJobContent(@HeaderParam("sessionid") String sessionId, @PathParam("jobid") String jobId)
+            throws NotConnectedRestException, UnknownJobRestException, PermissionRestException,
+            SubmissionClosedRestException, JobCreationRestException {
+        try {
+            Scheduler s = checkAccess(sessionId, "/scheduler/jobs/" + jobId + "/xml");
+            final Job jobContent = s.getJobContent(JobIdImpl.makeJobId(jobId));
+            return new Job2XMLTransformer().jobToxml((TaskFlowJob) jobContent);
+        } catch (NotConnectedException e) {
+            throw new NotConnectedRestException(e);
+        } catch (UnknownJobException e) {
+            throw new UnknownJobRestException(e);
+        } catch (PermissionException e) {
+            throw new PermissionRestException(e);
+        } catch (SubmissionClosedException e) {
+            throw new SubmissionClosedRestException(e);
+        } catch (JobCreationException e) {
+            throw new JobCreationRestException(e);
+        } catch (TransformerException | ParserConfigurationException e) {
+            throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)